addons/binding/org.openhab.binding.km200/src/main/java/org/openhab/binding/km200/internal/KM200SwitchProgramService.java

```diff
@@ -0,0 +1,570 @@
+                    }
+                }
+            }
+        }
+        HashMap<String, ArrayList<Integer>> weekMap = switchMap.get(setpoint);
+        if (weekMap == null) {
+            initWeeklist(setpoint);
+            weekMap = switchMap.get(setpoint);
+        }
+        ArrayList<Integer> dayList = weekMap.get(day);
+        dayList.add(time);
+        Collections.sort(dayList);
+    }
+
+    /**
+     * This function removes all switches from the switchmap
+     *
+     */
+    void removeAllSwitches() {
+        if (switchMap != null) {
+            switchMap.clear();
+        }
+    }
+
+    void setMaxNbOfSwitchPoints(Integer nbr) {
+        if (nbr != null) {
+            maxNbOfSwitchPoints = nbr;
+        }
+    }
+
+    void setMaxNbOfSwitchPointsPerDay(Integer nbr) {
+        if (nbr != null) {
+            maxNbOfSwitchPointsPerDay = nbr;
+        }
+    }
+
+    void setSwitchPointTimeRaster(Integer raster) {
+        if (raster != null) {
+            switchPointTimeRaster = raster;
+        }
+    }
+
+    void setSetpointProperty(String property) {
+        setpointProperty = property;
+    }
+
+    /**
+     * This function sets the day
+     *
+     */
+    void setActiveDay(String day) {
+        if (!days.contains(day)) {
+            logger.error("This type of weekday is not supported, get day: {}", day);
+            throw new IllegalArgumentException("This type of weekday is not supported, get day: " + day);
+        }
+        activeDay = day;
+    }
+
+    /**
+     * This function sets the cycle
+     *
+     */
+    void setActiveCycle(Integer cycle) {
+        if (cycle > this.getMaxNbOfSwitchPoints() / 2 || cycle > this.getMaxNbOfSwitchPointsPerDay() / 2 || cycle < 1) {
+            logger.error("The value of cycle is not valid, get cycle: {}", cycle);
+            throw new IllegalArgumentException("The value of cycle is not valid, get cycle: " + cycle.toString());
+        }
+        /* limit the cycle to the next one after last (for creating a new one) */
+        if (cycle > (getNbrCycles() + 1) || getNbrCycles() == 0) {
+            activeCycle = getNbrCycles() + 1;
+        } else {
+            activeCycle = cycle;
+        }
+    }
+
+    /**
+     * This function sets the positive switch to the selected day and cycle
+     *
+     */
+    void setActivePositiveSwitch(Integer time) {
+        Integer actTime;
+        if (time < MIN_TIME) {
+            actTime = MIN_TIME;
+        } else if (time > MAX_TIME) {
+            actTime = MAX_TIME;
+        } else {
+            actTime = time;
+        }
+        synchronized (switchMap) {
+            HashMap<String, ArrayList<Integer>> week = switchMap.get(getPositiveSwitch());
+            if (week != null) {
+                ArrayList<Integer> daysList = week.get(getActiveDay());
+                if (daysList != null) {
+                    Integer actC = getActiveCycle();
```

Comment:
Please do not abbreviate this far, it makes code less readable.

Reply:
If you do not intend to and handle null's please use the primitive types ie. int boolean etc.